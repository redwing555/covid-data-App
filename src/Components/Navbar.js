import React from 'react';
import { useHistory, useLocation } from 'react-router-dom';
import { BsChevronLeft } from 'react-icons/bs';
// import { FaMicrophone } from 'react-icons/fa';
// import { IoIosSettings } from 'react-icons/io';
import '../styles/navbar.css';

const Navbar = () => {
  const history = useHistory();
  const location = useLocation();

  return (
    <nav>

      <div className="location-path">
        {(location.pathname !== '/') ? <button className="back" type="button" onClick={history.goBack}><BsChevronLeft className="fas fa-chevron-left"> </BsChevronLeft></button> : <span className="back">Countries</span>}
        {' '}
        <span className="location">

          {location.pathname}
        </span>
      </div>

      {/* <div className="icons">
        <div className="icon"><FaMicrophone className="mx-2" /></div>
        <div className="icon"><IoIosSettings className="mx-2" /></div>

      </div> */}

    </nav>
  );
};

export default Navbar;
