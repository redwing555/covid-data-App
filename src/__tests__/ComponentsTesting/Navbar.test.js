import React from 'react';
import renderer from 'react-test-renderer';
import { HashRouter, Route } from 'react-router-dom';

import Navbar from '../../Components/Navbar';

describe('Navigation', () => {
  it('renderers route correctly', () => {
    const tree = renderer
      .create(
        <HashRouter>

          <Route path="/country">
            <Navbar />
          </Route>

        </HashRouter>
        ,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
