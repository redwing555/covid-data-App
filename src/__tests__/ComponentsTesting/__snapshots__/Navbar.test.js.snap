// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation renderers route correctly 1`] = `null`;
